--- mysql-connector-python-2.1.3/lib/cpy_distutils.py~
+++ mysql-connector-python-2.1.3/lib/cpy_distutils.py
@@ -131,7 +131,7 @@ def unix_lib_is64bit(lib_file):
     stdout = prc.communicate()[0]
     stdout = stdout.split(':')[1]
     log.debug("# lib_file {0} stdout: {1}".format(lib_file, stdout))
-    if 'x86_64' in stdout or 'x86-64' in stdout or '32-bit' not in stdout:
+    if 'x86_64' in stdout or 'x86-64' in stdout and '32-bit' not in stdout:
         return True
 
     return False
@@ -213,7 +213,7 @@
                 "be executed: {0}".format(str(exc)))
 
         if stdout:
-            if '64' in stdout:
+            if '64' in stdout and not '32-bit' in stdout:
                 info['arch'] = "x86_64"
             else:
                 info['arch'] = "i386"
