--- mysql-connector-python-8.0.16/lib/cpy_distutils.py~	2019-04-26 06:57:29.000000000 +0200
+++ mysql-connector-python-8.0.16/lib/cpy_distutils.py	2019-04-26 06:58:18.098036824 +0200
@@ -183,7 +183,7 @@ def unix_lib_is64bit(lib_file):
     stdout = prc.communicate()[0]
     stdout = stdout.split(':')[1]
     log.debug("# lib_file {0} stdout: {1}".format(lib_file, stdout))
-    if 'x86_64' in stdout or 'x86-64' in stdout or '32-bit' not in stdout:
+    if ('x86_64' in stdout or 'x86-64' in stdout) and '32-bit' not in stdout:
         return True
 
     return False
