diff -ur mysql-connector-python-1.1.6.org/python2/mysql/connector/connection.py mysql-connector-python-1.1.6/python2/mysql/connector/connection.py
--- mysql-connector-python-1.1.6.org/python2/mysql/connector/connection.py	2014-02-11 22:37:14.000000000 +0100
+++ mysql-connector-python-1.1.6/python2/mysql/connector/connection.py	2014-02-26 14:43:44.219350687 +0100
@@ -296,7 +296,7 @@
             if key == 'ssl_verify_cert':
                 set_ssl_flag = True
                 self._ssl.update({'verify_cert': value})
-            elif key.startswith('ssl_') and value:
+            elif key.startswith('ssl_'):
                 set_ssl_flag = True
                 self._ssl.update({key.replace('ssl_', ''): value})
             else:
diff -ur mysql-connector-python-1.1.6.org/python3/mysql/connector/connection.py mysql-connector-python-1.1.6/python3/mysql/connector/connection.py
--- mysql-connector-python-1.1.6.org/python3/mysql/connector/connection.py	2014-02-11 22:37:14.000000000 +0100
+++ mysql-connector-python-1.1.6/python3/mysql/connector/connection.py	2014-02-26 14:43:53.192986440 +0100
@@ -295,7 +295,7 @@
             if key == 'ssl_verify_cert':
                 set_ssl_flag = True
                 self._ssl.update({'verify_cert': value})
-            elif key.startswith('ssl_') and value:
+            elif key.startswith('ssl_'):
                 set_ssl_flag = True
                 self._ssl.update({key.replace('ssl_', ''): value})
             else:
