--- mysql-connector-python-8.0.11/tests/mysqld.py~	2018-04-10 14:35:30.000000000 +0200
+++ mysql-connector-python-8.0.11/tests/mysqld.py	2018-05-28 08:25:18.652587431 +0200
@@ -184,35 +184,7 @@ class MySQLServerBase(object):
         Raises MySQLBootstrapError when something fails.
         """
 
-        # Locate mysqld, mysql binaries
-        LOGGER.info("Locating mysql binaries (could take a while)")
-        files_to_find = [EXEC_MYSQL, EXEC_MYSQLD]
-        for root, dirs, files in os.walk(self._basedir):
-            if self._sbindir:
-                break
-            for afile in files:
-                if (afile == EXEC_MYSQLD and
-                        os.access(os.path.join(root, afile), 0)):
-                    self._sbindir = root
-                    LOGGER.debug("Located {} in {}".format(
-                        EXEC_MYSQLD, self._sbindir))
-                    try:
-                        files_to_find.remove(EXEC_MYSQLD)
-                    except ValueError:
-                            pass
-                elif (afile == EXEC_MYSQL and
-                        os.access(os.path.join(root, afile), 0)):
-                    self._bindir = root
-                    LOGGER.debug("Located {} in {}".format(
-                        EXEC_MYSQL, self._bindir))
-                    try:
-                        files_to_find.remove(EXEC_MYSQLD)
-                    except ValueError:
-                            pass
-
-                if not files_to_find:
-                    break
-
+        self._sbindir = '/usr/sbin'
 
         if not self._sbindir:
             raise MySQLBootstrapError(
